src/main/java/ca/on/oicr/pineryreports/reports/impl/OctaneCountsReport.java

```diff
@@ -0,0 +1,487 @@
+        throw new ParseException("After date must be in format yyyy-mm-dd");
+      }
+      this.start = after;
+    }
+    
+    if (cmd.hasOption(OPT_BEFORE)) {
+      String before = cmd.getOptionValue(OPT_BEFORE);
+      if (!before.matches(DATE_REGEX)) {
+        throw new ParseException("Before date must be in format yyyy-mm-dd");
+      }
+      this.end = before;
+    }
+  }
+
+  @Override
+  public String getTitle() {
+    return "OCTANE Counts "
+        + (start == null ? "Any Time" : start)
+        + " - "
+        + (end == null ? "Now" : end)
+        + ", and Inventory "
+        + new SimpleDateFormat("yyyy-MM-dd").format(new Date());
+  }
+  
+  private static final String ATTR_CATEGORY = "Sample Category";
+  private static final String ATTR_TISSUE_TYPE = "Tissue Type";
+  private static final String ATTR_TISSUE_ORIGIN = "Tissue Origin";
+  private static final String ATTR_SLIDES = "Slides";
+  private static final String ATTR_DISCARDS = "Discards";
+  private static final String ATTR_CONSUMED = "Slides Consumed";
+  private static final String ATTR_REMAINING = "Remaining";
+  private static final String ATTR_STAIN = "Stain";
+  
+  private static final String SAMPLE_CLASS_SLIDE = "Slide";
+  private static final String SAMPLE_CLASS_WHOLE_RNA = "whole RNA";
+  
+  private static final String SAMPLE_CATEGORY_IDENTITY = "Identity";
+  private static final String SAMPLE_CATEGORY_TISSUE = "Tissue";
+  private static final String SAMPLE_CATEGORY_STOCK = "Stock";
+  private static final String SAMPLE_CATEGORY_ALIQUOT = "Aliquot";
+
+  private static final String STAIN_HE = "Hematoxylin+Eosin";
+  
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
+    List<SampleDto> allOctaneSamples = pinery.getSample().allFiltered(new SamplesFilter().withProjects(Lists.newArrayList("OCT")));
+    Map<String, SampleDto> allOctaneSamplesById = mapSamplesById(allOctaneSamples);
+    
+    List<SampleDto> buffyCoats = new ArrayList<>(); // Ly_R tissues
+    List<SampleDto> plasma = new ArrayList<>(); // Pl_R tissues
+    List<SampleDto> ctDnaPlasma = new ArrayList<>(); // Ct_T tissues
+    List<SampleDto> slides = new ArrayList<>(); // Slides
+    List<SampleDto> stocksFromBuffy = new ArrayList<>(); // Ly_R stocks
+    List<SampleDto> stockDnaFromSlides = new ArrayList<>(); // DNA Stocks with Slide in hierarchy
+    List<SampleDto> stockRnaFromSlides = new ArrayList<>(); // RNA Stocks with Slide in hierarchy
+    List<SampleDto> dnaAliquots = new ArrayList<>(); // DNA aliquots
+    List<SampleDto> dnaAliquotsFromSlides = new ArrayList<>(); // DNA aliquots with Slide in hierarchy
+    List<SampleDto> dnaAliquotsFromBuffy = new ArrayList<>(); // DNA aliquots with Ly_R tissue
+    List<SampleDto> rnaAliquots = new ArrayList<>(); // RNA aliquots
+    List<SampleDto> rnaAliquotsFromSlides = new ArrayList<>(); // RNA aliquots with Slide in hierarchy
+    
+    for (SampleDto sam : allOctaneSamples) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)